pandas/tests/indexing/test_loc.py

```diff
@@ -711,3 +711,42 @@ def test_identity_slice_returns_new_object(self):
 
         original_series[:3] = [7, 8, 9]
         assert all(sliced_series[:3] == [7, 8, 9])
+
+    @pytest.mark.parametrize(
+        'indexer_type_1',
+        (list, tuple, set, slice, np.ndarray, Series, Index))
+    @pytest.mark.parametrize(
+        'indexer_type_2',
+        (list, tuple, set, slice, np.ndarray, Series, Index))
+    def test_loc_getitem_nested_indexer(self, indexer_type_1, indexer_type_2):
+        # GH #19686
+        # .loc should work with nested indexers which can be
+        # any list-like objects (see `pandas.api.types.is_list_like`) or slices
+
+        def convert_nested_indexer(indexer_type, keys):
+            if indexer_type == np.ndarray:
+                return np.array(keys)
+            if indexer_type == slice:
+                return slice(*keys)
+            return indexer_type(keys)
+
+        a = [10, 20, 30]
+        b = [1, 2, 3]
+        index = pd.MultiIndex.from_product([a, b])
+        df = pd.DataFrame(np.arange(len(index)), index=index, columns=['Data'])
```

Comment:
you need to add ``dtype='int64'`` otherwise this won't match on windows (as np.arange defaults to int32)

Reply:
good catch, thanks